tabsie/server.R

```diff
@@ -255,18 +257,24 @@ shinyServer(
     output$freqTable <- renderTable({#validation done before this is called, no need to repeat
       if (!valAuth) return;#break processing of not authorized.
       pdata = getpData(input$filter, serverDataDic, serverData)
-      addmargins(table(pdata[,c(input$xVal,input$yVal)]))
-    })
+      if(input$xVal==input$yVal) {
+        data.frame(Counts=cbind(summary(pdata[,input$xVal])))
+      } else {
+        as.data.frame.matrix(addmargins(table(pdata[,c(input$xVal,input$yVal)])))
+      }
+    },rownames=T)
```

Comment:
Note the `rownames=T` outside the expression block. Finally figured out how to do that. The unlabeled rows were driving me nuts.